### PR TITLE
Pull Request into Master for added enhancements

### PR DESCRIPTION
**Pull request by JeremyH011**
This should add the requested enhancements #3, #11 and #12 .

Issues that should be left are related to each other #13, #14 . I will continue to look into this.

### CODE REVIEW

File: src/extend_quiz.py [hunk 1]
```diff
@@ -107,13 +112,14 @@ def getStudents(course_id, auth_header):
...
-	df = pd.DataFrame(columns=['name','SIS_id','canvas_id','extra_time'])
+	df = pd.DataFrame(columns=['name','SIS_id','canvas_id','extra_time','extra_attempts'])
 	for student in student_list:
 		df = df.append({
 			'name':student['name'],
 			'SIS_id': student['sis_user_id'],
 			'canvas_id': student['id'],
-			'extra_time': 0
+			'extra_time': 0,
+			'extra_attempts': 0
```
Comment (alisonmyers) [inline src/extend_quiz.py:122]: set default as null

Comment (JeremyH011) [reply src/extend_quiz.py:122]: I don't think the API will be very happy if we pass a null value into it when used later. Reason I was hesitant on doing so, and the fact that the zero doesn't add any extra time or attempts.

Comment (JeremyH011) [reply src/extend_quiz.py]: Default values have been set to null (Python is None) now.

**Comment by JeremyH011** [comment]
Added error printing based on quiz-reports, and error logging.

**Comment by JeremyH011** [comment]
Added documentation in README.md and Jupyter notebook file that should address issues #13 and #14 .

**Pull request merged**